README.md

```diff
@@ -377,6 +377,34 @@ okta:
     rateLimit:
       maxRetries: 4
 ```
+
+When you use OAuth 2.0, the full YAML configuration would look like:
+
+```yaml
+okta:
+  client:
+    connectionTimeout: 30 # seconds
+    orgUrl: "https://{yourOktaDomain}" # i.e. https://dev-123456.oktapreview.com
+    proxy:
+      port: null
+      host: null
+      username: null
+      password: null
+    authenticationScheme: "OAUTH2"
+    authorizationMode: "PrivateKey"
+    clientId: "{clientId}"
+    scopes:
+    - okta.clients.read
+    - okta.clients.manage
+    privateKey:
+        # PEM format is recommended.
+        # This SDK supports following RSA AND EC algorithms - RS256, RS384, RS512, ES256, ES384, ES512.
+        keyFilePath: "~/.okta/privateKey.pem"
+        algorithm: "RS256"
```

Comment:
This does go against what we have set up for our spec. https://github.com/okta/oss-technical-designs/blob/master/technical_designs/oauth-in-management-sdks.md#privatekey-format-to-be-defined-by-each-sdk-the-recommended-format-is-pem states `Format to be defined by each SDK` meaning providing a `pem` file, inline pem, or jwk.  This should be updated to be 

```
privateKey: "~/.okta/privateKey.pem"
```

You should be able to determine the algorithm in Java by using some library.  In general, we want the developer to provide as little detail as possible in this config, that way they don't have to figure out the algorithm themselves and potentially run into errors if they get it wrong.

Reply:
👍 @bretterer I missed that in the spec
We could probably check for the armor lines, and fall back to a file.

This also fixes one of the oddities of the property to env var mapping:
`okta.client.privateKey.keyFilePath` -> `OKTA_CLIENT_PRIVATE_KEY_KEY_FILE_PATH` (note the `KEY_KEY`)

Reply:
Noted, will update.

Reply:
Updated code and pushed. More tests to come in next push.

Reply:
Pushed more tests.